## Seminar 7

Vi skal her se på caset med bryggeriene Graff Brygghus og Bryggeri 13, og se om det vil lønne seg for disse å inngå et samarbeid.
Informasjon om markedet: 
I markedet for mikroøl har vi to bryggerier i Tromsø, Graff Brygghus og Bryggeri 13. I produksjon av øl velger bedriftene produksjonsvolum i god tid før produktene er på markedet. Etterspørselen i markedet er gitt ved:
	P = 175 – 2Q
hvor P er markedspris for mikroøl per flaske, Q er totalt kvantum (antall tusen flasker) og Q = qG + qB, der qG er produsert kvantum for Graff Brygghus og qB er produsert kvantum for Bryggeri 13. I produksjon av mikroøl vil Graff Brygghus og Bryggeri 13 ha konstante marginalkostnader på kr 10 per flaske, og begge bedriftene har årlige faste kostnader på kr 500 000.


Ved Cournot konkurranse i dette markedet vil optimal produksjon for hvert av bryggeriene være på 27 500 flasker, til en pris på kr 65 per flaske. Dette vil gi bedriftene en profitt på 1012.500.
Anta at bryggeriene blir enige om å koordinere sin adferd. De avtaler å sette monopolpris og å dele markedet likt mellom seg. Videre avtaler de at hvis en av dem i en periode avviker fra denne avtalen, vil de begge i alle framtidige perioder opptre som konkurrenter.

a)	Hva vil total produksjon av øl være hvis bryggeriene velger samarbeid og hvor mye vil bryggeriene tjene på dette?

b)	Hvor mye vil Graff Brygghus tjene på å avvike fra samarbeidsavtalen med Bryggeri 13 i en perioe?

c)	Hvor stor må diskonteringsfaktoren være for at bryggeriene skal ha incentiver til å opprettholde samarbeidsavtalen oavtalen?


In [2]:
import sympy as sp
from sympy import *
import numpy as np
from matplotlib import pyplot as plt

Monopoltilpasning

In [3]:
def demand(Q):
    return (175-2*Q)

In [4]:
def marginalrevenue(Q):
    return (175-4*Q)

In [5]:
def marginalcost(Q):
    return (10)

In [6]:
# Optimal tilpasning der MR = MC
Q=sp.symbols('Q', real=True, positive=True)
equ=sp.Eq(marginalrevenue(Q),marginalcost(Q))
equ

Eq(175 - 4*Q, 10)

In [7]:
# Optimalt kvantum for monopol
Q_equ=sp.solve(equ,Q)[0]
Q_equ
round(Q_equ,2)

41.25

In [8]:
# Optimalt kvantum for hvert bryggeri
print (Q_equ/2)
round(Q_equ/2,3)

165/8


20.625

In [9]:
# Optimal pris
p_eq=demand(Q_equ)
p_eq
round(p_eq,2)

92.50

In [10]:
# Profitt for monopoltilpasning
def profitt(Q):
    return (demand(Q).subs({Q:Q_equ})-10)*Q_equ
round(profitt(Q),1)

3403.1

In [11]:
# Profitt ved samarbeid for hvert av bryggeiene
def profitt(Q):
    return ((demand(Q).subs({Q:Q_equ})-10)*Q_equ)/2-500
round(profitt(Q),1)

1201.6

## Optimal tilpasning ved Cournot modell

In [12]:
# Vi har symmetirske bedrifer hvor løsningen blir qG = qB = q*
def demand_G(qG):
           return (175-2*qG-2*qB)

In [13]:
def demand_B(qB):
           return ((175-2*qG-2*qB))

In [14]:
def marginalrevenue_G(qG):
    return (175-4*qG-2*qB)

In [15]:
def marginalrevenue_B(qB):
    return (175-2*qG-4*qB)


In [16]:
# marginalkostnaden for begge bedriftene er lik 10 
qG=sp.symbols('qG', real=True, positive=True)
qB=sp.symbols('qB', real=True, positive=True)
equ=sp.Eq(marginalrevenue_B(qB),10)
equ

Eq(-4*qB - 2*qG + 175, 10)

In [17]:
#reaksjonsfunksjon til Bryggeri 13
qB_equ=sp.solve(equ,qB)[0]
qB_equ

165/4 - qG/2

In [18]:
qG=sp.symbols('qG', real=True, positive=True)
equ=sp.Eq(marginalrevenue_G(qG),10)
equ

Eq(-2*qB - 4*qG + 175, 10)

In [19]:
#reaksjonsfunksjon til Graff bryggerier
qG_equ=sp.solve(equ,qG)[0]
qG_equ

165/4 - qB/2

### Optimal tilpasning ved avvik når Graff Brygghus produserer avtalt mengde på 41.25/2 = 20.625 og Bryggeri 13 velger å avvike fra avtalen om samarbeid

In [20]:
# Bryggeri 13 vil produsere følgende:
qB_equ.subs({qG:Q_equ/2})


495/16

In [21]:
round(qB_equ.subs({qG:Q_equ/2}),3)

30.938

In [22]:
# prisen i markedet blir:
demand_B(qB).subs({qG:Q_equ/2,qB:qB_equ})


575/8

In [23]:
round(demand_B(qB).subs({qG:Q_equ/2,qB:qB_equ}),3)

71.875

In [33]:
# profitt til Graff Brygghus er lik (p -c)qG -500:
round((demand_B(qB).subs({qG:Q_equ/2,qB:qB_equ})-10)*(Q_equ/2)-500,3)

776.172

In [32]:
# profitt til Bryggeri 13 blir:
round((demand_B(qB).subs({qG:Q_equ/2,qB:qB_equ})-10)*(qB_equ.subs({qG:Q_equ/2}))-500,3)

1414.258

## Gjentatte spill

For å se om prissamarbeid vil lønnne seg ved gjentatte spill må vi sammenligne nåverdien av samarbeid (deling av profitt ved monopoltilpasning) med nåverdien ved avvik (profitt ved avvik i en periode og profitt ved Cournot-tilpasning resten av periodene)

In [1]:
from sympy import Symbol
from sympy.solvers.inequalities import reduce_rational_inequalities

In [2]:
x = Symbol('x', real=True)

In [3]:
reduce_rational_inequalities([[x + 2 > 0]], x)

-2 < x

In [4]:
# Nåverdi ved sammarbeid > nåverdi ved avvik - med p=1
reduce_rational_inequalities([[1201.6/(1-x) > 1414.258+(x*1012.5)/(1-x)]], x)

(0.529318644557171 < x) & (x < 1.0)

In [5]:
r = Symbol('r', real=True)

In [9]:
reduce_rational_inequalities([[1/(1+r) > 0.52931]], r)

(-1.0 < r) & (r < 0.889252045115339)

In [10]:
# Nåverdi ved sammarbeid > nåverdi ved avvik - med p=0.6
reduce_rational_inequalities([[1800/(1-(0.6*x)) > 2025+((0.6*x)*1600)/(1-(0.6*x))]], x)

(0.882352941176471 < x) & (x < 1.66666666666667)

In [11]:
reduce_rational_inequalities([[1/(1+r) > 0.88235]], r)

(-1.0 < r) & (r < 0.133337111123704)